packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -461,11 +462,37 @@ export default function useAutocomplete(props) {
     },
   );
 
+  const previousProps = usePreviousProps({
+    filteredOptions,
+    value,
```

Comment:
I think `value` should not be attached again to ref to store the value. You can use `prevValue.current` from above.

Reply:
I've removed `prevValue.current` and kept value in `usePreviousProps` since usePreviousProps is being used in code.